### ISSUE TITLE
transitive empty starter failed at parsing

### ISSUE THREAD

**Issue by b3b00** [b3b00/csly#507]
```c#

public enum OptionTestToken
    {
        [Lexeme("a")] a = 1       
    }

 public class TransitiveEmptyStarterParser
    {
        [Production("x : y")]
        public string X(string y)
        {
            return y;
        }

        [Production("y : z*")]
        public string Y(List<string> zs)
        {
            if (zs.Any())
            {
                return string.Join(",", zs);
            }

            return "empty";
        }

        [Production("z : a")]
        public string Z(Token<OptionTestToken> a)
        {
            return a.Value;
        }
    }
```

this parser failed at parsing an empty string that should succeed : y can be empty as its only clause is a zero or more that can be empty.

**Comment by b3b00** [comment]
rvealed by stryker analysis : 
![image](https://github.com/user-attachments/assets/9dc8a81f-2bf3-44f6-9d81-a2977c5f1aa5)
https://github.com/b3b00/csly/blob/98664028f6223b0aafb6dfdfc435f3c36596c0e8/src/sly/parser/parser/llparser/ebnf/EBNFRecursiveDescentSyntaxParserStarter.cs#L40

**Comment by b3b00** [comment]
fix with #508
